ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,109 @@ actor_system_config:
+```
+
+### Soft memory limit {#soft-memory-limit}
+
+The soft memory limit specifies a dangerous threshold that should not be exceeded under normal circumstances.
+
+If the soft limit is exceeded, {{ ydb-short-name }} starts to reduce the shared cache size to zero. Therefore, more database nodes should be added to the cluster as soon as possible, or per-component memory limits should be reduced.
+
+### Target memory utilization {#target-memory-utilization}
+
+The target memory utilization specifies a threshold for memory usage that is considered optimal.
+
+Flexible cache sizes are calculated according to their limit thresholds to keep process consumption around this value.
+
+For example, in a database that consumes a little memory on query execution, caches consume memory around this threshold, and other memory stays free. If query execution consumes more memory, caches start to reduce their limits to their minimum threshold.
+
+### Per-component memory limits
+
+#### Flexible memory limits
+
+Some {{ ydb-short-name }} components have both minimum and maximum memory limit thresholds, allowing for dynamic adjustment based on current process consumption.
+
+These components are:
+
+- Shared cache
+- MemTable
+
+Each of these components' limits is dynamically recalculated every second so that each component consumes memory proportionally to its limit threshold, and the total consumed memory stays around the target memory utilization.
+
+These components' minimum memory limit threshold isn't reserved and remains free until consumed.
+
+Memory limits can be configured either in absolute bytes or as a percentage relative to the [hard memory limit](#hard-memory-limit). Using percentages is advantageous for managing clusters with nodes of varying capacities. If both absolute byte and percentage limits are specified, the memory controller uses a combination of both (maximum for lower limits and minimum for upper limits).
+
+If needed, both the minimum and maximum thresholds should be overridden; otherwise, a missing threshold will have a default value.
+
+Example of the `memory_controller_config` section with specified shared cache limits:
+
+```yaml
+memory_controller_config:
+  shared_cache_min_percent: 10
+  shared_cache_max_percent: 30
+```
+
+#### Non-flexible memory limits
+
+Other {{ ydb-short-name }} components have only a fixed memory limit without minimum and maximum thresholds.
+
+These components are:
+
+- KQP
+
+The memory limit for these components indicates the amount of memory a component may try to consume. Still, consumption can be rejected if a database node exceeds the soft memory limit. Therefore, the sum of these limits may exceed the soft memory limit, but each individual limit is expected to be less than the soft memory limit.
+
+Example of the `memory_controller_config` section with a specified KQP limit:
+
+```yaml
+memory_controller_config:
+  query_execution_limit_percent: 25
+```
+
+### Configuration
```

Comment:
It's not clear to a reader what kind of rules she has to obey. Sum of all params must be 100?